clang/lib/CIR/Lowering/DirectToLLVM/LowerToLLVM.cpp

```diff
@@ -2311,6 +2311,38 @@ static mlir::Value createLLVMBitOp(mlir::Location loc,
   return result;
 }
 
+class CIRObjSizeOpLowering
+    : public mlir::OpConversionPattern<mlir::cir::ObjSizeOp> {
+public:
+  using OpConversionPattern<mlir::cir::ObjSizeOp>::OpConversionPattern;
+
+  mlir::LogicalResult
+  matchAndRewrite(mlir::cir::ObjSizeOp op, OpAdaptor adaptor,
+                  mlir::ConversionPatternRewriter &rewriter) const override {
+    auto resTy =
+        getTypeConverter()->convertType(op.getType()).cast<mlir::IntegerType>();
+    auto loc = op->getLoc();
+
+    std::string llvmIntrinName = "llvm.objectsize";
+    auto llvmIntrinNameAttr =
+        mlir::StringAttr::get(rewriter.getContext(), llvmIntrinName);
+    // kindInfo can only be 0 or 2, a.k.a min or max,
```

Comment:
If you are unsure, use `FIXME` or `TODO` when writing comments for things that need to be tackled later.

Reply:
You can delete the comment: mix/max are enforced by the verifier, so no other value could show up here. Also, lowering the intrinsic call only accepts both, see `CIRGenBuiltin.cpp:808`.